Makefile

```diff
@@ -13,34 +30,57 @@ TARGETARCH=$(shell gcc -dumpmachine)
 #
 # Added the following flags:
 # -fsanitize=address,undefined # enable ASan/UBSan
-CFLAGS = -O2 -Wall -Wformat -Wformat=2 -Wconversion -Wimplicit-fallthrough \
-  -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 \
-	-fstack-clash-protection \
-	-fstack-protector-strong -Wl,-z,nodlopen -Wl,-z,noexecstack -Wl,-z,relro \
-	-Wl,-z,now -Wl,--as-needed -Wl,--no-copy-dt-needed-entries -Wtrampolines \
-	-Wbidi-chars=any -fPIE -pie -Werror=implicit \
-	-Werror=incompatible-pointer-types -Werror=int-conversion \
-	-fno-delete-null-pointer-checks -fno-strict-overflow -fno-strict-aliasing \
-	-fsanitize=undefined
-
-ifeq ($(TARGETARCH), x86_64-linux-gnu)
-CFLAGS += -fcf-protection=full # only supported on x86_64
+WARN_CFLAGS := -Wall -Wformat -Wformat=2 -Wconversion -Wimplicit-fallthrough \
+	-Werror=format-security -Werror=implicit -Werror=int-conversion \
+	-Werror=incompatible-pointer-types
+
+ifneq (,$(findstring gcc,$(CC_VERSION)))
+WARN_CFLAGS += -Wtrampolines -Wbidi-chars=any  # clang as for 18.1.8 doesn't support this warnings
```

Comment:
Sounds too hacky for my taste... Since as for now the only other compiler that's in the picture is clang, let's just disable the warnings for it... If anybody will need to support other compilers it would be easier to switch to a build system from this century like cmake.

done